### ISSUE TITLE
Does this work for SharePoint 2016??

### ISSUE THREAD

**Issue by tommls** [mganss/RegexReplaceProvider#1]
Can this be used for a Windows Server 2016 IIS 8.5 SharePoint 2016 site??
If you don't know, what is your recommendation for the best way to try this so things don't break??
How does one UN-install this if it does not work??
Thank you, Tom

**Comment by mganss** [comment]
Yes, although I have no experience with Sharepoint. My recommendation is to try this on a local machine. If things don't work, simply remove the rule and provider from your URL Rewrite configuration (all in IIS Manager). There is no additional uninstall step necessary, because the provider is not installed into IIS, i.e. once you have removed the rule and provider from your IIS configuration, the provider DLL is just another file on your machine (IIS remains untainted so to speak).

**Comment by tommls** [comment]
The provided gacutil.ps1 script did not work even though it SAID it installed the dll...it never showed in the dropdown.
I used the script in this URL to get the dll installed and appearing in the dropdown:
https://www.andrewcbancroft.com/2015/12/16/using-powershell-to-install-a-dll-into-the-gac/
If you have any suggestions for newbies on how to use the URL rewrite module, that would be nice, though I will research for myself too, the single sentence you wrote was not something I could figure out since I've never used IIS url rewrite before.
Thank you, Tom

**Comment by tommls** [comment]
I read your readme several times and I understand I could make an url rewrite rule with just this in it??
example/{SpaceToHyphen:{R:1}}
What I don't understand is why the 'example/' is in there.
I want to only replace spaces or %20 with hyphens, regardless where they occur.
Thank you for your patience, Tom

**Comment by mganss** [comment]
If you're new to IIS Rewrite, please read [Creating Rewrite Rules for the URL Rewrite Module](https://docs.microsoft.com/en-us/iis/extensions/url-rewrite-module/creating-rewrite-rules-for-the-url-rewrite-module) first. In the example, `example/{SpaceToHyphen:{R:1}}` would go into the rule *action*, i.e. the definition of what the URL should look like after rewriting. The `{R:1}` is a backreference to the first group in the rule pattern (see docs) and `SpaceToHyphen` replaces spaces with hyphens in that part. `example/` is just to illustrate that the "call" to `SpaceToHyphen` can occur anywhere within the action, e.g. if you wanted to rewrite URLs like `OldSite/Example Page.aspx` as `newsite/example-page.aspx` you'd match the `Example Page.aspx` part as a regex group in your rule pattern and put `newsite/{SpaceToHyphen:{R:1}}` into the rule action.

**Comment by tommls** [comment]
Good explanation, I will read it 4-6 times more, and I will do that reading you recommend. My urls don't contain the word 'example' so I imagine I would just type {SpaceToHyphen:{R:1}}. I'll try that and test etc. Thank you for your patience, Tom
